kyaml/fn/runtime/runtimeutil/runtimeutil_test.go

```diff
@@ -1512,3 +1513,53 @@ metadata:
 		assert.Equal(t, tc.expected, *NewContainerEnvFromStringSlice(fn.Container.Env))
 	}
 }
+
+func TestPrintResults(t *testing.T) {
+	resNode, _ := yaml.Parse(`
+name: search-replace
+items:
+  - message: Mutated field value to "my-nginx"
+    severity: info
+    resourceRef:
+      apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        namespace: my-space
+        name: my-deployment
+    field:
+      path: spec.selector.matchLabels.app
+    file:
+      path: path/to/deployment.yaml
+  - message: Mutated field value to "my-nginx"
+    severity: info
+    resourceRef:
+      apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        namespace: my-space
+        name: my-deployment
+    field:
+      path: spec.template.metadata.labels.app
+    file:
+      path: path/to/deployment.yaml
+  - message: Mutated field value to "my-nginx"
+    severity: info
+    resourceRef:
+      apiVersion: v1
+      kind: Service
+      metadata:
+        namespace: my-space
+        name: my-svc
+    field:
+      path: spec.selector.app
+    file:
+      path: path/to/svc.yaml`)
+
+	out := &bytes.Buffer{}
+	err := printResults(resNode, out)
+	assert.NoError(t, err)
+	assert.Equal(t, `[info] path/to/deployment.yaml apps/v1/Deployment/my-space/my-deployment spec.selector.matchLabels.app: Mutated field value to "my-nginx"
```

Comment:
This is not very readable. A more readable presentation:

https://github.com/GoogleContainerTools/kpt-functions-sdk/blob/b7e346f0c18f972edf7ca02272734fbc68fca905/ts/kpt-functions/src/types.ts#L430

Only different is we use `resource` instead of `object`.

Reply:
Also many of the fields in Result are optional, need to test those scenarios